client/src/views/Projects.vue

```diff
@@ -36,11 +44,50 @@ export default defineComponent({
       store.dispatch.loadProject(project);
     };
 
+    // this array represents the strings returned by the API
+    const scanStates = ['complete', 'unreviewed', 'needs tier 2 review'];
+    let scanStateCounts = computed(() => Object.fromEntries(scanStates.map(
+      (stateString) => [stateString, 0],
+    )));
+    const getScanStateCounts = () => {
+      if (currentProject.value) {
+        scanStateCounts = computed(() => Object.fromEntries(scanStates.map(
+          (stateString) => {
+            const stateCount = Object.entries(currentTaskOverview.value.scan_states).filter(
```

Comment:
I'm not sure I understand what you mean. Do you mean we should calculate the current states for each scan on the backend and allow the frontend to retrieve that information?

If we calculate the current states for each scan on the backend, then the frontend would need to make a call to update the donut / scan list every time a `ScanDecision` is submitted. If we can compute scan states on the frontend from what the `store` already knows about scans, it eliminates what would be a repetitive call. When we submit a `ScanDecision`, it updates what the `store` knows about that `Scan`, and this gets recomputed on a new render. 

Correct me if I'm misinterpreting your suggestion.

Reply:
I missed that `scan_states` was also used in switchReviewMode. I though that this was the only place it was consumed, and all that you do with it here is some filter/mapping to get the state counts, which is something that could instead be done on the backend. Since `scan_states` is used elsewhere, I think the way you're doing things here makes sense.